### PR TITLE
RD-5500 Fix fetch URLs in deployment-related input types

### PR DESCRIPTION
**Pull request by qooban**
## Description

This PR fixes bug described in RD-5500 by providing valid fetch URLs for input fields requiring deployment ID.
Additionally it fixes one more bug related to that fetch URL (see more details in code comments).

Once merged, this change will be cherry-picked to `6.4.0-build` branch.

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
1. [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3335/) - only 2 specs - PASSED
2. [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3344/) - all specs - one test case failed in Getting Started spec. I checked it locally and it's passing. Failure looks unrelated (flaky test?), so I'm considering it as PASSED

## Documentation
N/A

### CODE REVIEW

File: widgets/common/src/inputs/fields/types.ts [hunk 1]
```diff
@@ -16,5 +16,7 @@ export interface ErrorAwareInputFieldProps extends InputFieldProps {
     error: InputProps['error'];
 }
 
-export type DynamicDropdownInputFieldProps = ErrorAwareInputFieldProps &
-    Pick<DynamicDropdownProps, 'toolbox' | 'constraints' | 'multiple'>;
+export type DynamicDropdownInputFieldProps = ErrorAwareInputFieldProps & { constraints: Constraint[] } & Pick<
```
Comment (qooban) [inline widgets/common/src/inputs/fields/types.ts:19]: Improved typing to be more strict and precise.

In `DynamicDropdown` the type for `constraints` is set to `Array<any>`, but `*InputField` components are always getting `constraints` prop as `Constraint[]`.

File: widgets/common/src/inputs/fields/NodeIdInputField.tsx [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import DynamicDropdown from '../../components/DynamicDropdown';
 import type { DynamicDropdownInputFieldProps } from './types';
+import useFetchUrlWithDeploymentId from './useFetchUrlWithDeploymentId';
 
 export default function NodeIdInputField({ name, onChange, toolbox, ...restProps }: DynamicDropdownInputFieldProps) {
-    const fetchUrl = `/searches/nodes?_include=id&deployment_id=${toolbox.getContext().getValue('deploymentId')}`;
+    const fetchUrl = useFetchUrlWithDeploymentId('/searches/nodes?_include=id', restProps.constraints);
```
Comment (qooban) [inline widgets/common/src/inputs/fields/NodeIdInputField.tsx:6]: There were 2 bugs here:
1. `deployment_id` was set in query string always no matter if there was deployment ID constraint used
2. `deployment_id` query param was taken from toolbox context which is not proper solution for all cases (Execute Workflow modal can be opened without `deploymentId` set in the toolbox context)

The first is described in RD-5500, the second I fixed at once without having dedicated ticket.

That comment applies to all deployment-related `*InputField` components.

File: widgets/common/src/inputs/fields/useFetchUrlWithDeploymentId.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { useContext, useMemo } from 'react';
+import DeploymentIdContext from '../utils/deploymentIdContext';
+import getConstraintValueFunction from '../utils/getConstraintValueFunction';
+import type { Constraint } from '../types';
+
+export default function useFetchUrlWithDeploymentId(fetchUrl: string, constraints: Constraint[]) {
```
Comment (qooban) [inline widgets/common/src/inputs/fields/useFetchUrlWithDeploymentId.ts:6]: This custom hook was created to create proper `fetchUrl` for deployment-related `*InputFields` components.

It returns `fetchUrl` with added `deployment_id` query param only if it is set in the React context (not toolbox context) and there is no `deployment_id` constraint defined.

File: widgets/common/src/inputs/utils/getConstraintValueFunction.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import type { Constraint } from '../types';
+
+export default function getConstraintValueFunction(constraints: Constraint[]) {
```
Comment (qooban) [inline widgets/common/src/inputs/utils/getConstraintValueFunction.ts:3]: Moved this function out from `InputField.tsx` to use it in `useFetchUrlWithDeploymentId.ts`.

**Review by kubama — APPROVED** [review]
Looks cool, I really like the solution :+1:

**Review by NorbertAmartus — APPROVED** [review]
LGTM 🎉 
Thanks for providing more details via additional comments 🚀

**Comment by qooban** [comment]
Thanks for approvals. Before merging, I have triggered another system tests run (all specs this time) to double check if that change does not brake anything.

**Pull request merged**